src/ocean/sys/Stats.d

```diff
@@ -236,3 +240,197 @@ public class CpuMemoryStats
+
+*******************************************************************************/
+
+public class GCStats
+{
+    static if ( hasMember!(GC, "monitor") )
+    {
+        /**********************************************************************
+
+            The type of the gcEnd delegate
+
+        **********************************************************************/
+
+        private alias ParameterTupleOf!(GC.monitor)[1] MonitorEndParams;
+
+        /**********************************************************************
+
+            Integers used by the gcEnd delegate
+
+        **********************************************************************/
+
+        private alias ParameterTupleOf!(MonitorEndParams)[0] MonitorInt;
+    }
+    else
+    {
+        /**********************************************************************
+
+            Default type for intgers used by monitor
+
+        **********************************************************************/
+
+        private alias long MonitorInt;
+    }
+
+    /**********************************************************************
+
+        Structure representing recorded GC stats.
+
+    **********************************************************************/
+
+    public struct Stats
+    {
+        /**********************************************************************
+
+            The number of microseconds for how much the garbage collector
+            was run.
+
+        **********************************************************************/
+
+        public size_t gc_run_duration;
+
+        /**********************************************************************
+
+            The percentage of time that was spent by the garbage collector.
+
+        **********************************************************************/
+
+        public float gc_run_percentage = 0;
+
+        /**********************************************************************
+
+            The number of bytes freed overall
+
+        **********************************************************************/
+
+        int freed;
+
+        /**********************************************************************
+
+            The number of bytes freed within full pages
+
+        **********************************************************************/
+
+        int pagebytes;
+    }
+
+    /**********************************************************************
+
+        ditto
+
+    **********************************************************************/
+
+    private Stats stats;
+
+    /**********************************************************************
+
+        Timestamp for when the last garbage collection started
+
+    **********************************************************************/
+
+    private ulong gc_start_us;
+
+    /**********************************************************************
+
+        Timestamp for when the last stats were written to hdd
```

Comment:
"Timestamp for when the last stats were collected" - there's nothing here implying the stats will go to the hard drive.

Reply:
I also don't get if this one is right. Doesn't this pretty much say "timestamp when the stats collecting was started or when it was reset"?

Reply:
(that being said, I doubt this is useful)

Reply:
If you insist, you could make it `last_collected_timestamp` and set it in `collect`

Reply:
(still, I would just remove this member, it doesn't bring anything IMHO and makes the stats more confusing).

Reply:
I updated the comment and the name. I did this pr in a hurry, so I forgot to use it. :P

It was needed to calculate the percentage time for how long the gc was run.

Reply:
How do you calculate the percentage of the time based on this? I think what you need is the time while the program is running, right?

Reply:
the duration is computed here https://github.com/sociomantic-tsunami/ocean/pull/661/files#diff-8e40a07cedca7d7b1b28be980b65f3faR211 and is for each collection cycle

Reply:
Yup, see my later comment: https://github.com/sociomantic-tsunami/ocean/pull/661#issuecomment-442044627

Reply:
Maybe just put `(used to calculate the percentage of the time spent in GC between stats cycles)

Reply:
I updated the comments and the release notes